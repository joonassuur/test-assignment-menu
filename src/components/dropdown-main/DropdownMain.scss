.dropdown-container {
  width: 100%;
  border: 1px solid #dcdce0;
  box-shadow: 0px 2px 8px rgba(51, 51, 51, 0.24);
  border-radius: 8px;
  max-width: 304px;
}
