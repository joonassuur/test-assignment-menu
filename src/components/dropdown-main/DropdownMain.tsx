import React from "react";
import { useSelector } from "react-redux";
import { getActiveDD } from "../../redux/Index";

import useComponentVisible from "../component-visible/ComponentVisible";
import DropdownGeneral from "./dropdown-general/DropdownGeneral";
import DropdownAchievements from "./dropdown-achievements/DropdownAchievements";

import "./DropdownMain.scss";

function DropdownMain() {
  const ref = useComponentVisible();

  const activeDD = useSelector(getActiveDD);

  const renderActiveDD = () => {
    if (activeDD === "achievements") {
      return <DropdownAchievements />;
    }
    return <DropdownGeneral />;
  };

  return (
    <div ref={ref} className="dropdown-container">
      {renderActiveDD()}
    </div>
  );
}

export default DropdownMain;
