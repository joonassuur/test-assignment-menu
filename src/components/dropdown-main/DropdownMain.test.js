import React from "react";
import TestRenderer from "react-test-renderer";

import DropdownMain from "./DropdownMain";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../redux/Index");

describe("DropdownMain component", () => {
  test("Matches the snapshot", () => {
    const dropdownMain = TestRenderer.create(<DropdownMain />);
    expect(dropdownMain.toJSON()).toMatchSnapshot();
  });
});
