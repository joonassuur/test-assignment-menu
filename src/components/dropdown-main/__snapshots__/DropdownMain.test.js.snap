// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMain component Matches the snapshot 1`] = `
<div
  className="dropdown-container"
>
  <div
    className="dropdown-general"
  >
    <div
      className="head"
    >
      Your Companies
    </div>
    <div
      className="companies-container"
    >
      <div
        className="item company-container "
        onClick={[Function]}
      >
        <div
          className="name"
        >
          <div
            className="title"
          >
            Name OÜ
          </div>
          <div
            className="subtitle"
          >
            Subtitle 1
          </div>
        </div>
        
      </div>
      <div
        className="item company-container "
        onClick={[Function]}
      >
        <div
          className="name"
        >
          <div
            className="title"
          >
            Name2 AS
          </div>
          <div
            className="subtitle"
          >
            
          </div>
        </div>
        
      </div>
      <div
        className="item company-container "
        onClick={[Function]}
      >
        <div
          className="name"
        >
          <div
            className="title"
          >
            Name3 AS
          </div>
          <div
            className="subtitle"
          >
            
          </div>
        </div>
        
      </div>
    </div>
    <div
      className="achievements-container item"
      onClick={[Function]}
    >
      <div
        className="achievements-sub-container-top"
      >
        <div
          className="settings-icon"
        >
          <span
            className="material-icons-outlined"
          >
            school
          </span>
        </div>
        <div
          className="achievements-info"
        >
          <div
            className="title"
          >
            Achievements
          </div>
          <div
            className="subtitle"
          >
            3/11 done 
            <span
              className="dot"
            >
              ·
            </span>
              27%
          </div>
        </div>
      </div>
      <div
        className="achievements-progress"
      >
        <progress
          className="progress"
          max="100"
          value="27"
        />
      </div>
    </div>
    <div
      className="otherlinks-container"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <span
          className="material-icons-outlined"
        >
          phone_iphone
        </span>
        <span>
          Get the mobile app
        </span>
      </div>
      <div
        className="item"
        onClick={[Function]}
      >
        <span
          className="material-icons-outlined"
        >
          people
        </span>
        <span>
          Community
        </span>
      </div>
      <div
        className="item"
        onClick={[Function]}
      >
        <span
          className="material-icons-outlined"
        >
          book
        </span>
        <span>
          Knowledge base
        </span>
      </div>
    </div>
    <div
      className="settings-logout-container"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <span
          className="material-icons-outlined"
        >
          settings
        </span>
        <span>
          Settings
        </span>
      </div>
      <div
        className="item exit"
        onClick={[Function]}
      >
        <span
          className="material-icons-outlined"
        >
          exit_to_app
        </span>
        <span>
          Log out
        </span>
      </div>
    </div>
  </div>
</div>
`;
